pkg/revision/interface.go

```diff
@@ -0,0 +1,7 @@
+package revision
+
+type Revision interface {
```

Comment:
Maybe the `Revision` interface and the relevant struct can be changed to `RevisionUpdater` because this interface contains less info about a revision, but more info about how to update a revision including start, stop, update